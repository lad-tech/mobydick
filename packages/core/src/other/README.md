## @lad-tech/mobydick-core/other

## ***Counter***

Счетчик с 1 до 99+

Пропсы:
type — primary, secondary, tertiary, tertiary, destructive;
count — число;
без count — будет нарисован indicator c цветом ElementError;
style — можно передать свой backgroundColor, и через top, right, bottom, left — расположить в нужном месте, по дефолту position: absolute;
size — medium(24), small(18), по default — medium;
maxLength —— по default 2, отображает максимальное заданное количество чисел;

## ***CrossedText***

Использовать, когда цвет зачеркнутой линии отличается от цвета текста, в противном случае используйте дефолтный пропс textDecorationLine: 'line-through'.

## ***Collapsible***

Не используйте `typeAnimation`: `spring` и `keyboard`, происходит краш приложения на android;


## ***Carousel***

isLoop – бесконечная карусель, вы должны понимать, что элементов в ней лучше немного использовать
isDots – не использовать, они еще в работе


## ***Dots***
fixedSize – будет работать, если элементов меньше 7
