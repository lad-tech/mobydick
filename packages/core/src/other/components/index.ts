export * from './Badge/';
export * from './CrossedText';
export * from './Dots';
export * from './Avatar';
export * from './Carousel';
export * from './Status';
export * from './Collapsible';
export * from './Rating';
