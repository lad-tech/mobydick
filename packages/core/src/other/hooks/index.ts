import useAnimatedShake from './useAnimatedShake';

export {useAnimatedShake};
