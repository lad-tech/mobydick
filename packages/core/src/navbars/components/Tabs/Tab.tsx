import {useCallback} from 'react';

import {ITabProps} from '../../types';
import useStyles from '../../../styles/hooks/useStyles';
import TouchableOpacity from '../../../basic/components/TouchableOpacity/TouchableOpacity';
import {TypographyLegacy} from '../../../typography/components/TypographyLegacy/TypographyLegacy';
import {LABELS} from '../../../other';

import stylesCreate from './stylesCreate';

const Tab = (props: ITabProps): JSX.Element => {
  const [styles, theme] = useStyles(stylesCreate);
  const {
    active,
    item,
    fontTab,
    fontActiveTab,
    backgroundColorTab,
    backgroundColorActiveTab,
    styleTab,
    styleActiveTab,
    onPressCommon,
  } = props;

  const backgroundColorActive =
    backgroundColorActiveTab || theme.colors.ElementBase;
  const backgroundColor = backgroundColorTab || theme.colors.BgTertiary;
  const font = fontTab || 'Regular-Tertiary-XS';
  const fontActive = fontActiveTab || 'Regular-White-XS';

  const selectPressable = useCallback(() => {
    if (item.onPress) {
      item.onPress();
    } else if (onPressCommon) {
      onPressCommon(item);
    }
  }, [item.onPress, onPressCommon]);

  return (
    <TouchableOpacity
      onPress={selectPressable}
      accessibilityLabel={LABELS.tab}
      style={[
        styles.tab,
        active ? styleActiveTab : styleTab,
        {
          backgroundColor: active ? backgroundColorActive : backgroundColor,
        },
      ]}>
      {item.leftIcon ? item.leftIcon : null}
      <TypographyLegacy font={active ? fontActive : font}>
        {item.label}
      </TypographyLegacy>
      {item.rightIcon ? item.rightIcon : null}
    </TouchableOpacity>
  );
};

export default Tab;
