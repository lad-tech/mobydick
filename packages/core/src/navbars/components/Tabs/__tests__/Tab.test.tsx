import {fireEvent, render} from '@testing-library/react-native';

import Tab from '../Tab';
import View from '../../../../basic/components/View/View';
import {LABELS} from '../../../../other';

describe('Tab', () => {
  const value = 1;
  const label = '1';

  test('render correctly with props', () => {
    const {toJSON, getByLabelText} = render(
      <Tab
        item={{
          label: label,
          value: value,
          onPress: () => null,
          leftIcon: <View />,
          rightIcon: <View />,
        }}
        active={true}
        backgroundColorTab={'#fff000'}
        fontTab={'Medium-Accent-M'}
      />,
    );
    const tab = getByLabelText(LABELS.tab);

    fireEvent.press(tab);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render correctly active false', () => {
    const {toJSON, getByLabelText} = render(
      <Tab
        item={{
          label: label,
          value: value,
        }}
        active={false}
      />,
    );
    const tab = getByLabelText(LABELS.tab);

    fireEvent.press(tab);
    expect(toJSON()).toMatchSnapshot();
  });
  test('render correctly active true', () => {
    const onPressCommon = jest.fn();
    const {getByLabelText} = render(
      <Tab
        item={{
          label: label,
          value: value,
        }}
        onPressCommon={onPressCommon}
        active={true}
      />,
    );
    const tab = getByLabelText(LABELS.tab);

    fireEvent.press(tab);
    expect(onPressCommon).toHaveBeenCalledWith({label, value});
  });
});
