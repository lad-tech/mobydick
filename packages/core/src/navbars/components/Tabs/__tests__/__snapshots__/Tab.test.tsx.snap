// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab render correctly active false 1`] = `
<View
  accessibilityLabel="tab"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#D0D6E0",
      "borderRadius": 8,
      "marginRight": 8,
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 6,
    }
  }
>
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#5E6678",
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 18,
          "minHeight": 18,
        },
        undefined,
      ]
    }
  >
    1
  </Text>
</View>
`;

exports[`Tab render correctly with props 1`] = `
<View
  accessibilityLabel="tab"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#2B78EE",
      "borderRadius": 8,
      "marginRight": 8,
      "opacity": 1,
      "paddingHorizontal": 12,
      "paddingVertical": 6,
    }
  }
>
  <View />
  <Text
    allowFontScaling={true}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "fontWeight": "400",
          "lineHeight": 18,
          "minHeight": 18,
        },
        undefined,
      ]
    }
  >
    1
  </Text>
  <View />
</View>
`;
